python/ray/tune/trainable/trainable.py

```diff
@@ -512,12 +515,18 @@ def _maybe_save_to_cloud(self, checkpoint_dir: str) -> bool:
             return True
 
         checkpoint = Checkpoint.from_directory(checkpoint_dir)
-        retry_fn(
-            lambda: checkpoint.to_uri(self._storage_path(checkpoint_dir)),
+        checkpoint_uri = self._storage_path(checkpoint_dir)
+        if not retry_fn(
+            lambda: checkpoint.to_uri(checkpoint_uri),
             subprocess.CalledProcessError,
             num_retries=3,
             sleep_time=1,
-        )
+            timeout=self.sync_timeout,
+        ):
+            logger.error(
+                f"Could not upload checkpoint even after 3 retries: "
+                f"{checkpoint_uri}"
+            )
```

Comment:
can we please teach people how to extend sync_timeout in the error message?
```Could not upload checkpoint after 3 retries. Current sync_timeout is {self.sync_timeout}, if you think this is not enough, you can extend it by setting SyncConfig.timeout
```
?